interface.h

```diff
@@ -64,82 +64,337 @@
 
+/**
+ * @brief Adds a new timed event. An existing event timer at the provided index
+ *        identifier will be replaced.
+ *
+ * @param index Used to register and identify the timed event. Valid values are
+ *        between 0 and TIMED_EVENT_COUNT - 1
+ * @param ms How often the timed event should be triggered in milliseconds
+ * @param repeat 0 or 1 value indicating if the timed event should repeat
+ *        indefinitely
+ */
 void add_timed_event(uint8_t index, uint16_t ms, uint8_t repeat);
+
+/**
+ * @brief Stops and removes a previously added timed event at the index
+ *        identifier.
+ * 
+ * @param index Index identifier of the timed event to stop
+ */
 void stop_timed_event(uint8_t index);
+
+/**
+ * @brief Updates a previously registed timed event trigger interval in
+ *        milliseconds
+ * 
+ * @param index Index identifier of the timed event to update
+ * @param ms How often the timed event should be triggered in milliseconds
+ */
 void update_timer_interval(uint8_t index, uint16_t ms);
 
 
 // ----------------------------------------------------------------------------
 // clock
 
+/**
+ * @brief Get the global time value of milliseconds since start
+ * 
+ * @return uint64_t Returns milliseconds since start
+ */
 uint64_t get_global_time(void);
+
+/**
+ * @brief Check if hardware has a CV clock input
+ * 
+ * @return uint8_t Returns 0 if hardware does not have CV clock input,
+ *         uint8_t Returns 1 if hardware has CV clock input
+ */
 uint8_t has_clock_input(void);
+
+/**
+ * @brief Check if external CV clock input is connected
+ * 
+ * @return uint8_t Returns 0 if external CV clock is not connected,
+ *         uint8_t Returns 1 if external CV clock is connected
+ */
 uint8_t is_external_clock_connected(void);
+
+/**
+ * @brief Set the clock output on or off
+ * 
+ * @param on 0 if clock output should be off,
+ *           1 if clock output should be on
+ */
 void set_clock_output(uint8_t on);
 
 
 // ----------------------------------------------------------------------------
 // inputs
 
+/**
+ * @brief Get the number of hardware CV inputs available
+ * 
+ * @return uint8_t Returns number of hardware CV inputs
+ */
 uint8_t get_cv_input_count(void);
+
+/**
+ * @brief Get the CV value for the provided input index.
+ * 
+ * Note: this functionality is not fully implemented. Teletype is currently the
+ * only Monome module that has a CV input. 
+ * 
+ * @param index Hardware CV input number (first input is index 0)
+ * @return int16_t Returns CV value represented as a signed integer with the
+ *         range of -16,384 (-10v) and 16,384 (+10v).
+ */
 int16_t get_cv(uint8_t index);
 
+/**
+ * @brief Get the number of hardware CV gate inputs available
+ * 
+ * @return uint8_t Returns number of hardware CV gate inputs
+ */
 uint8_t get_gate_input_count(void);
+
+/**
+ * @brief Get the CV gate value for the provided input index.
+ * 
+ * Note: this functionality is only functional for CV gate input 0 of the
```

Comment:
should work for all inputs, for teletype you can have up to 8 gate inputs (i'll need to double check if one input is always dedicated to be the clock input)

Reply:
Ah that makes sense. I made the note because of this: https://github.com/scanner-darkly/multipass/blob/0b6cb34e40e9d8ced92ac7258da6972593bf0ca2/monome_euro/main.c#L1009

Reply:
ah, if i recall correctly that's because `handler_tr` is not used by teletype.. i'll check when i do more work on multipass later this week!